Documentation/gettingstarted/egress-gateway.rst

```diff
@@ -6,160 +6,401 @@
+configure egress gateway policies to route and SNAT the egress traffic for a
+specific workload.
 
-   This guide assumes that Cilium has been correctly installed in your
-   Kubernetes cluster. Please see :ref:`k8s_quick_install` for more
-   information. If unsure, run ``cilium status`` and validate that Cilium is up
-   and running.
+.. note::
 
-Enable Egress Gateway
+    This guide assumes that Cilium has been correctly installed in your
+    Kubernetes cluster. Please see :ref:`k8s_quick_install` for more
+    information. If unsure, run ``cilium status`` and validate that Cilium is up
+    and running.
+
+Preliminary Considerations
+==========================
+
+Cilium must make use of network-facing interfaces and IP addresses present on
+the designated gateway nodes. These interfaces and IP addresses must be
+provisioned and configured by the operator based on their networking
+environment. The process is highly-dependent on said networking environment. For
+example, in AWS/EKS, and depending on the requirements, this may mean creating
+one or more Elastic Network Interfaces with one or more IP addresses and
+attaching them to instances that serve as gateway nodes so that AWS can
+adequately route traffic flowing from and to the instances. Other cloud
+providers have similar networking requirements and constructs.
+
+Additionally, the enablement of the egress gateway feature requires that both
+BPF masquerading and the kube-proxy replacement are enabled, which may not be
+possible in all environments (due to, e.g., incompatible kernel versions).
+
+Compatibility with other features
+=================================
+
+L7 policies
+-----------
+
+Egress gateway is currently partially incompatible with L7 policies.
+Specifically, when an egress gateway policy and an L7 policy both select the same
+endpoint, traffic from that endpoint will not go through egress gateway, even if
+the policy allows it. Full support will be added in an upcoming release once
+:gh-issue:`19642` is resolved.
+
+LB acceleration
+---------------
+
+When the egress gateway feature is used in combination with XDP-based LB
+acceleration (``--bpf-lb-acceleration=native``), the user must ensure that the
+host Iptables configuration allows packets through the ``FORWARD`` chain. Full
+support will be added in an upcoming release once :gh-issue:`19717` is resolved.
+
+Enable egress gateway
 =====================
 
-The feature is disabled by default. The egress gateway requires BPF
-masquerading, which itself requires BPF NodePort to be enabled. An easy way to
-enable all requirements is as follows.
+The egress gateway feature and all the requirements can be enabled as follow:
 
 .. tabs::
-
     .. group-tab:: Helm
 
-        If you installed Cilium via ``helm install``, you may enable
-        the Egress gateway feature with the following command:
+        .. code-block:: shell-session
+
+            $ helm upgrade cilium |CHART_RELEASE| \\
```

Comment:
You need this block to be a `.. parsed-literal::`, or the `|CHART_RELEASE|` substitution won't happen.